runner.py

```diff
@@ -15,28 +15,29 @@ def inner():
         return res
     return inner
 
-move = _refresh(_board.move)
-turn_left = _refresh(_board.turn_left)
-pick_beeper = _refresh(_board.pick_beeper)
-put_beeper = _refresh(_board.put_beeper)
-front_is_clear = _board.front_is_clear
-right_is_clear = _board.right_is_clear
-left_is_clear = _board.left_is_clear
-beeper_is_present = _board.beeper_is_present
-facing_north = _board.facing_north
-facing_south = _board.facing_south
-facing_east = _board.facing_east
-facing_west = _board.facing_west
-
 def execute(callback):
     try:
         _board.start_screen()
         _board.draw()
         callback()
         return callback
-    except LogicException, e:
+    except LogicException as e:
         _board.draw_exception(e)
-    except Exception, e:
+    except Exception as e:
         raise e
-    finally:
-        _board.end_screen()
```

Comment:
This line here was the main reason for adding `execute`, and for using it as a decorator on some sort of `main` function. Without calling `curses.endwin()`, my bash session gets left in a bad state. I looked into some ways to get that same functionality without needing to introduce the language learner to decorators (since you're right that it's preferable to not have to add one more new unexplained thing).

One approach is to add the following after we create `_board` in `runner.py`.
```python
@atexit.register
def end_curses():
    _board.end_screen()
```